src/Model/Export/Catalog/ProductField/GenericField.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Omikron\Factfinder\Model\Export\Catalog\ProductField;
+
+use Magento\Catalog\Model\Product;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\Phrase;
+use Magento\Framework\ObjectManagerInterface;
+use Omikron\Factfinder\Api\Export\Catalog\ProductFieldInterface;
+use Omikron\Factfinder\Exception\AttributeExportException;
+use Psr\Log\LoggerInterface;
+
+class GenericField implements ProductFieldInterface
+{
+    /** @var string */
+    private $attributeName;
+
+    /** @var ScopeConfigInterface */
+    private $scopeConfig;
+
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(
+        string $attributeName,
+        ScopeConfigInterface $scopeConfig,
+        ObjectManagerInterface $objectManager
+    ) {
+        $this->attributeName = $attributeName;
+        $this->scopeConfig   = $scopeConfig;
+        $this->logger        = $objectManager->get('Omikron\Factfinder\Export\Logger');
+    }
+
+    /**
+     * @param Product $product
+     *
+     * @return string
+     * @throws AttributeExportException
+     */
+    public function getValue(Product $product): string
+    {
+        try {
+            return (string) $product->getAttributeText($this->attributeName);
```

Comment:
This one throws a Fatal Error, i don't think it's catchable. I'd rather check for attribute existence. Also, we have to mimic what we already do for the Attributes field, i wouldnt restrict it to attributes with option.

Reply:
i moved whole switch instruction from `Attributes` to separate class and use it also in `GenericField`. Now it should be capable of exporting different types of attributes.